src/impy/models/sibyll.py

```diff
@@ -64,32 +85,24 @@ def __init__(self, event_kinematics, seed=None, logfname=None):
 
         # This calls _set_event_kinematics which uses self._lib.isib_pdg2pid
         # which works only after an initialization call to self._lib.pdg_ini()
-        self.event_kinematics = event_kinematics
+        self.kinematics = evt_kin
 
         self._set_final_state_particles()
 
-    def _cross_section(self, evt_kin):
-        if evt_kin is None:
-            evt_kin = self.event_kinematics
-
-        sigproj = None
-        if abs(evt_kin.p1pdg) in [2212, 2112]:
-            sigproj = 1
-        elif abs(evt_kin.p1pdg) == 211:
-            sigproj = 2
-        elif abs(evt_kin.p1pdg) == 321:
-            sigproj = 3
-        else:
-            raise ValueError(
-                f"No cross section available for projectile {evt_kin.p1pdg}"
+    def _cross_section(self):
+        kin = self.kinematics
+        if kin.p2.A > 1:
+            # TODO figure out what this returns exactly:
+            # self._lib.sib_sigma_hnuc
```

Comment:
It should be ok to use. The docs says:

output:      SIGprod    particle production cross section (mb)
C                  SIGbdif    q.ela and ela beam diff. cross section
C                  SIGela     elastic cross section

Reply:
I moved this to an issue, so that we can address it in another PR.